### PR TITLE
👽️✨ #659: add GET /wizard/pdf endpoint to data mapper

### PR DESCRIPTION
**Pull request by samrichca**
## Description of Changes

This PR adds a GET `/wizard/pdf` endpoint to data mapper, so we can collect all the data for consent PDFs in one endpoint.

### Data Mapper
- Add GET `/wizard/pdf/:participant` endpoint to wizard router

### Types
- Add `GetConsentPdfDataResponse` schema and response type

## Testing

Tested with swagger as adult, guardian, and substitute.

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/data-mapper/src/routers/wizard.ts [hunk 1]
```diff
@@ -117,4 +117,75 @@ router.get('/progress/:participantId', async (req, res) => {
...
+		switch (consentPdfData.status) {
+			case 'SUCCESS': {
+				return res.status(200).json(consentPdfData.data);
+			}
+			case 'INVALID_REQUEST': {
+				return res
+					.status(400)
+					.json(ErrorResponse(REQUEST_VALIDATION_ERROR, consentPdfData.message));
+			}
+			case 'PARTICIPANT_DOES_NOT_EXIST': {
```
Comment (anncatton) [inline apps/data-mapper/src/routers/wizard.ts:174]: can add `case 'SIGNATURE_DOES_NOT_EXIST'` to the 404 response clause

**Review by anncatton — COMMENTED** [review]
Looks nice and tidy! Hopefully my comments on changing the `optional` fields are not me remembering the dashboard behaviour incorrectly, i think the intent of the PDF download is to only be available once the wizard has been completed + submitted?
Other comments are just for clarity/documentation.

File: packages/types/src/services/dataMapper/responses/GetConsentPdfData.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	createdAt: z.string().datetime().optional(),
+	currentLifecycleState: LifecycleState,
+	guardianName: OptionalName,
+	guardianRelationship: OptionalName,
+	participantOhipFirstName: Name,
+	participantOhipLastName: Name,
+	[RECONTACT__FUTURE_RESEARCH]: z.boolean().optional(),
+	[RECONTACT__SECONDARY_CONTACT]: z.boolean().optional(),
+	[RESEARCH_PARTICIPATION__CONTACT_INFORMATION]: z.boolean().optional(),
+	[RESEARCH_PARTICIPATION__FUTURE_RESEARCH]: z.boolean().optional(),
```
[the remark this answers is not in the document]
Comment (samrichca) [reply packages/types/src/services/dataMapper/responses/GetConsentPdfData.ts:50]: if they're not optional - I should add a step on this endpoint (or the consent API equivalent?) where I check that the lifecycle state is `CONSENTED`?

[the remark this answers is not in the document]
Comment (anncatton) [reply packages/types/src/services/dataMapper/responses/GetConsentPdfData.ts:50]: we have a check on the dashboard itself (or we will?) to check the current lifecycle state of the user, which would then show/enable the download pdf button, correct? Also, i expect the user would be able to download their pdf info in other states (just not `REGISTERED`).

[the remark this answers is not in the document]
Comment (samrichca) [reply packages/types/src/services/dataMapper/responses/GetConsentPdfData.ts]: OK, removed `optional` except for guardian fields.

File: apps/data-mapper/src/services/search.ts [hunk 3]
```diff
@@ -770,8 +781,8 @@ export const getConsentPdfData = async (
 		if (signaturesResult.data.length) {
 			const mostRecentSignature = signaturesResult.data[0];
 			consentReviewSign = {
-				createdAt: mostRecentSignature.createdAt,
 				signature: mostRecentSignature.signature,
+				signatureCreatedAt: mostRecentSignature.createdAt,
```
Comment (anncatton) [inline apps/data-mapper/src/services/search.ts:785]: 👍

**Review by anncatton — APPROVED** [review]
looks great! thanks @samrichca

**Pull request merged**